pages/data-providers/index.jsx

```diff
@@ -74,14 +85,39 @@ const getHelperMessage = ({ query, created, duplicated, error }) => {
   }
 }
 
-const DataProviderPage = () => {
+const DataProviderPage = ({ store }) => {
   const router = useRouter()
+  const [showSuccessMessage, setShowSuccessMessage] = useState(false)
+  const [formRef, dataProvider, resetDataProvider] = useDataProviderController()
+
+  useEffect(() => {
+    const handleRouteChange = (url) => {
+      if (url.startsWith('/data-providers')) {
+        const newUrl = new URL(url, window.location)
+        const urlParams = new URLSearchParams(newUrl.search)
+        if (urlParams.get('success') && store.dataProvider.created)
+          setShowSuccessMessage(true)
+        else {
+          setShowSuccessMessage(false)
+          resetDataProvider()
+        }
+      }
+    }
+
+    router.events.on('routeChangeStart', handleRouteChange)
+
+    return () => {
+      router.events.off('routeChangeStart', handleRouteChange)
+    }
+  }, [])
+
   const handleSubmitForm = () => {
-    router.push('/data-providers/finalisation')
+    router.push({
+      pathname: router.pathname,
+      query: { success: true },
```

Comment:
We should not change the URL. Add a boolean state for handling this. I can accidentally navigate here if press 'enter', see the issue below.

However, we should not change the URL because it should not be allowed to send the URL for the final page.

Reply:
It's not allowed. There is a check for it. Anyway, I am confused. When I add state to url you are against it and when I don't add it you say it would be nice to have it. 😄

Reply:
I am against the status state in the URL because the status should be checked. However, I support having the actual query in the URL but I am struggling with the parameter name. It's fine now

Reply:
I am not sure what do you mean. The status is checked.